File: clients/asset-client/src/constants/LoaiTaiLieuContractConstant.ts
import { createConstant } from "./Constant";

// Mã loại tài liệu đính kèm áp dụng cho Hợp đồng Chứng thực (AuthenticationContract).
// Tham chiếu spec: Documents/Hệ thống online.docx — section "Chứng thực hợp đồng điện tử".
const LoaiTaiLieuContractConstant = createConstant(
  {
    ChungMinhTenMien: "ChungMinhTenMien",
    DangKyChungThuc: "DangKyChungThuc",
    HuongDanDichVu: "HuongDanDichVu",
  } as const,
  {
    ChungMinhTenMien: { displayName: "Tài liệu chứng minh sở hữu tên miền" },
    DangKyChungThuc: { displayName: "Tài liệu đăng ký chứng thực" },
    HuongDanDichVu: { displayName: "Tài liệu hướng dẫn dịch vụ" },
  }
);

export default LoaiTaiLieuContractConstant;

File: API/Hinet.Service/Constant/LoaiTaiLieuContractConstant.cs
using System.ComponentModel;

namespace Hinet.Service.Constant
{
    // Mã loại tài liệu đính kèm cho Hợp đồng Chứng thực (AuthenticationContract).
    // Mirror FE: Client/src/constants/LoaiTaiLieuContractConstant.ts
    public class LoaiTaiLieuContractConstant
    {
        [DisplayName("Tài liệu chứng minh sở hữu tên miền")]
        public static string ChungMinhTenMien => "ChungMinhTenMien";

        public static readonly HashSet<string> ALL = new()
        {
            ChungMinhTenMien,
        };

        public static bool IsValid(string? code) =>
            !string.IsNullOrWhiteSpace(code) && ALL.Contains(code);
    }
}
